packages/trackerless-network/src/logic/EntryPointDiscovery.ts

```diff
@@ -119,20 +71,13 @@ export class EntryPointDiscovery {
         }
     }
 
-    async storeSelfAsEntryPointIfNecessary(currentEntrypointCount: number): Promise<void> {
+    async storeAndKeepSelfAsEntryPoint(): Promise<void> {
         if (this.abortController.signal.aborted) {
             return
         }
-        const possibleNetworkSplitDetected = this.config.layer1Node.getNeighborCount() < NETWORK_SPLIT_AVOIDANCE_LIMIT
-        if ((currentEntrypointCount < ENTRYPOINT_STORE_LIMIT) || possibleNetworkSplitDetected) {
-            this.isLocalNodeStoredAsEntryPoint = true
-            await this.storeSelfAsEntryPoint()
-            await this.keepSelfAsEntryPoint()
-        }
-        if (possibleNetworkSplitDetected) {
-            // TODO should we catch possible promise rejection?
-            setImmediate(() => this.avoidNetworkSplit())
-        }
+        this.isLocalNodeStoredAsEntryPoint = true
```

Comment:
Could we move the setting of this field into `storeSelfAsEntryPoint`?

Reply:
The point is to keep it here as the node will be stored as entry point until it leaves the stream. `keepSelfAsEntryPoint` would call `storeSelfAsEntryPoint `, reassigning the value which would be pointless.